pkg/providers/tinkerbell/tinkerbell.go

```diff
@@ -263,19 +263,23 @@ func (p *Provider) DatacenterConfig(_ *cluster.Spec) providers.DatacenterConfig
 func (p *Provider) MachineConfigs(_ *cluster.Spec) []providers.MachineConfig {
 	configs := make(map[string]providers.MachineConfig, len(p.machineConfigs))
 	controlPlaneMachineName := p.clusterConfig.Spec.ControlPlaneConfiguration.MachineGroupRef.Name
-	p.machineConfigs[controlPlaneMachineName].Annotations = map[string]string{p.clusterConfig.ControlPlaneAnnotation(): "true"}
-	if p.clusterConfig.IsManaged() {
-		p.machineConfigs[controlPlaneMachineName].SetManagedBy(p.clusterConfig.ManagedBy())
+	if _, ok := p.machineConfigs[controlPlaneMachineName]; ok {
```

Comment:
I don't think we want to silently ignore the fact a machine config is missing. I think we can prevent this from ever happening by validating all machine configs are present in `NewProvider()`.

Specifically, on [L117](https://github.com/aws/eks-anywhere/blob/main/pkg/providers/tinkerbell/tinkerbell.go#L117) we `TinkebrellMachineConfig` spec but we don't handle it not being present.